src/views/ErrorView.tsx

```diff
@@ -1,56 +1,58 @@
 import { useMemo } from "react";
 import { Layout } from "../components/Layout";
+import { useTranslation, Trans } from "react-i18next";
 import { useQueryParams } from "../utils/useQueryParams";
 
-const ERROR_CODES = {
-  mtls_fail: {
-    title: "Virheellinen mTLS-avain",
-    details:
-      "Selaimen mielestä mTLS-avaimesi on virheellinen (tai sitä ei ole). Jos juuri asensit avaimen laitteellesi, kokeile sulkea selain ja käynnistää se uudelleen.",
-  },
-  fall_back: {
-    title: "Virhe",
-    details: "Tapahtui virhe, jolle ei ole vielä virheilmoitusta.",
-  },
-};
-
 export function ErrorView() {
+  const { t, i18n } = useTranslation();
   const params = useQueryParams();
 
   const errorDetails = useMemo(() => {
     const code = params.get("code");
-    if (!code || !(code in ERROR_CODES)) return ERROR_CODES["fall_back"];
+    if (!code)
+      return {
+        titleKey: "error.fall_back.title",
+        detailsKey: "error.fall_back.details",
+      };
+
+    const titleKey = `error.${code}.title`;
+    const detailsKey = `error.${code}.details`;
 
-    return ERROR_CODES[code as keyof typeof ERROR_CODES];
+    return { titleKey, detailsKey };
```

Comment:
Should have default error, even when code doesn't match.

Reply:
now better?